evalml/automl/engines/sequential_engine.py

```diff
@@ -0,0 +1,68 @@
+from evalml.automl.engines import EngineBase
+
+
+class SequentialEngine(EngineBase):
+    """The default engine for the AutoML search. Evaluates pipelines locally and pipeline batches sequentially."""
+    def __init__(self):
+        super().__init__()
+        self.name = "Sequential Engine"
+
+    def evaluate_batch(self, pipeline_batch, result_callback=None, log_pipeline=False):
+        """ Evaluates a batch of pipelines in order. 
+
+        Arguments:
+            pipeline_batch (list(PipelineBase)): A batch of pipelines to be fitted and evaluated.
+            result_callback (callable): Function called once the pipeline is finished evaluating to store the results. If None, results will only be returned.
+            log_pipeline (bool): If True, log the pipeline and relevant information before evaluation.
+
+        Returns:
+            list(PipelineBase), list(dict), list(PipelineBase): A list of evaluated pipelines, the results of the evaluated pipelines, and the remaining pipelines in the batch that were not evaluated. 
+        """
+        super().evaluate_batch()
```

Comment:
Calling these dataset checks aren't super intuitive. Should I split them out into another function (like `check_search_info` or something like that)?